app/src/sharedTest/kotlin/io/eugenethedev/taigamobile/manager/TaigaTestInstanceManager.kt

```diff
@@ -112,12 +112,19 @@ class TaigaTestInstanceManager(
             .build()
 
         try {
-            // just wait for the server to actually warm up
-            while (checkRequest.execute().code >= 500) {
-                println("Instance is not ready yet, waiting 1s before another try")
-                Thread.sleep(1000)
+            // healthcheck - repeatedly (max 5 times) wait for the server to warm up with 1s interval
+            repeat(5) {
+                checkRequest.execute().let {
```

Comment:
Maybe something like this(trivia IMHO):
```
with(checkRequest.execute()) {
                    if (code >= 500) {
                        println("Instance is not ready yet, waiting 1s before another try")
                        Thread.sleep(1000)
                    } else {
                        successOrThrow()
                        return
                    }
                }
```

Reply:
I don't see any reasons to use `with` instead of `let` here since it won't affect readability